v03_pipeline/lib/misc/family_entries.py

```diff
@@ -0,0 +1,154 @@
+import hail as hl
+
+from v03_pipeline.lib.model import DatasetType
+
+
+def compute_callset_family_entries_ht(
+    dataset_type: DatasetType,
+    mt: hl.MatrixTable,
+    entries_fields: dict[str, hl.Expression],
+) -> hl.Table:
+    sample_id_to_family_guid = hl.dict(
+        {
+            s: family_guid
+            for family_guid, sample_ids in hl.eval(mt.family_samples).items()
+            for s in sample_ids
+        },
+    )
+    ht = mt.select_rows(
+        filters=mt.filters.difference(dataset_type.excluded_filters),
+        family_entries=(
+            # NB: we're sorted by both family and sample when this runs.
+            # However, the sort is not guaranteed once the entries
+            # table is editted and families are spliced out and re-appended.
+            hl.sorted(
+                hl.agg.collect(
+                    hl.Struct(
+                        s=mt.s,
+                        family_guid=sample_id_to_family_guid[mt.s],
+                        **entries_fields,
+                    ),
+                )
+                .group_by(lambda e: e.family_guid)
+                .values()
+                .map(
+                    lambda fe: hl.sorted(fe, key=lambda e: e.s),
+                ),
+                lambda fe: fe[0].family_guid,
+            )
+        ),
+    ).rows()
+    # NB: globalize before we set families to missing
+    ht = globalize_ids(ht)
+    ht = ht.annotate(
+        family_entries=(
+            ht.family_entries.map(
+                lambda fe: hl.or_missing(
+                    fe.any(dataset_type.family_entries_filter_fn),
+                    fe,
+                ),
+            )
+        ),
+    )
+    # Only keep rows where at least one family is not missing.
+    return ht.filter(ht.family_entries.any(lambda fe: ~hl.is_missing(fe)))
```

Comment:
nitpick, but I feel like `hl.is_defined` is cleaner than `~hl.is_missing`, especially because you then no longer need the lambda